src/plugins/gpunetio/gpunetio_backend.cpp

```diff
@@ -0,0 +1,1575 @@
+        result = DOCA_ERROR_CONNECTION_ABORTED;
+        return NIXL_ERR_BACKEND;
+    }
+
+    if (remote_conn_details_len_notif <= 0 || remote_conn_details_len_notif >= (size_t)-1) {
+        NIXL_ERROR << "Received wrong remote connection details, client, notif";
+        result = DOCA_ERROR_NO_MEMORY;
+        return NIXL_ERR_BACKEND;
+    }
+
+    remote_conn_details_notif = calloc (1, remote_conn_details_len_notif);
+    if (remote_conn_details_notif == NULL) {
+        NIXL_ERROR << "Failed to allocate memory for remote connection details";
+        result = DOCA_ERROR_NO_MEMORY;
+        return NIXL_ERR_BACKEND;
+    }
+
+    if (recv (oob_sock_client, remote_conn_details_notif, remote_conn_details_len_notif, 0) < 0) {
+        NIXL_ERROR << "Failed to receive remote connection details";
+        result = DOCA_ERROR_CONNECTION_ABORTED;
+        return NIXL_ERR_BACKEND;
+    }
+
+    //Avoid duplicating RDMA connection to the same QP by client/server threads
+    connectLock.lock();
+
+    if (connMap.find(remote_agent) != connMap.end()) {
+        NIXL_INFO << "QP for " << remote_agent << " already connected" << std::endl;
+        connectLock.unlock();
+        return NIXL_SUCCESS;
+    }
+
+    /* Connect local rdma to the remote rdma */
+    NIXL_INFO << "Connect DOCA RDMA to remote RDMA -- data" << std::endl;
+    result = doca_rdma_connect (rdma_qp->rdma_data,
+                                remote_conn_details_data,
+                                remote_conn_details_len_data,
+                                rdma_qp->connection_data);
+    if (result != DOCA_SUCCESS) {
+        NIXL_ERROR << "Function doca_rdma_connect data failed " << doca_error_get_descr (result);
+        free (remote_conn_details_data);
+        remote_conn_details_data = NULL;
+        connectLock.unlock();
+        return NIXL_ERR_BACKEND;
+    }
+
+    free (remote_conn_details_data);
+    remote_conn_details_data = NULL;
+
+    /* Connect local rdma to the remote rdma */
+    NIXL_INFO << "Connect DOCA RDMA to remote RDMA -- notif" << std::endl;
+    result = doca_rdma_connect (rdma_qp->rdma_notif,
+                                remote_conn_details_notif,
+                                remote_conn_details_len_notif,
+                                rdma_qp->connection_notif);
+    if (result != DOCA_SUCCESS) {
+        NIXL_ERROR << "Function doca_rdma_connect notif failed " << doca_error_get_descr (result);
+        free (remote_conn_details_notif);
+        remote_conn_details_notif = NULL;
+        connectLock.unlock();
+        return NIXL_ERR_BACKEND;
+    }
+
+    free (remote_conn_details_notif);
+    remote_conn_details_notif = NULL;
+
+    connMap[remote_agent] = 1;
+
+    connectLock.unlock();
+
+    return NIXL_SUCCESS;
+}
+
+nixl_status_t
+nixlDocaEngine::recvRemoteAgentName (int oob_sock_client, std::string &remote_agent) {
+    size_t msg_size;
+    char *remote_agent_;
```

Comment:
This is not needed. Use resize method of the string, and pass remote_agent.data() to recv.